kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/GenericKubernetesResource.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.api.model;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@JsonDeserialize(using = com.fasterxml.jackson.databind.JsonDeserializer.None.class)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({
+  "apiVersion",
+  "kind",
+  "metadata",
+  "items",
+})
+@Getter
+@Setter
+@ToString
+@EqualsAndHashCode
+public class GenericKubernetesResource implements HasMetadata {
+
+  @JsonProperty("apiVersion")
+  private String apiVersion;
+  @JsonProperty("kind")
+  private String kind;
+  @JsonProperty("metadata")
+  private ObjectMeta metadata;
+  @JsonIgnore
+  private transient Map<String, Object> additionalProperties = new HashMap<>();
```

Comment:
Can we use a LinkedHashMap here instead?  For the mock server logic I'll be able to manipulate GenericKubernetesResource objects directly and that will keep the output json looking very similar to the input.

Reply:
I think we can set the default value to LinkedHashMap.
However, I'm not sure if we can guarantee the same for deserialization purposes.

Reply:
I tested locally using 
private transient LinkedHashMap<String, Object> additionalProperties = new LinkedHashMap<>(); 
and everything worked as expected.